Source/Embeddings.Processing/CompareProjectionDefinitionsForAllTenants.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dolittle.Runtime.ApplicationModel;
+using Dolittle.Runtime.DependencyInversion;
+using Dolittle.Runtime.Lifecycle;
+using Dolittle.Runtime.Embeddings.Store.Definition;
+using Dolittle.Runtime.Tenancy;
+using System;
+
+namespace Dolittle.Runtime.Embeddings.Processing
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="ICompareEmbeddingDefinitionsForAllTenants" />.
+    /// </summary>
+    [Singleton]
+    public class CompareEmbeddingDefinitionsForAllTenants : ICompareEmbeddingDefinitionsForAllTenants
+    {
+        readonly IPerformActionOnAllTenants _onTenants;
+        readonly FactoryFor<IEmbeddingDefinitions> _getDefinitions;
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="CompareEmbeddingDefinitionsForAllTenants" /> class.
+        /// </summary>
+        /// <param name="onTenants">The tool for performing an action on all tenants.</param>
+        /// <param name="getDefinitions">The factory for getting Embedding definitions.</param>
+        public CompareEmbeddingDefinitionsForAllTenants(
+            IPerformActionOnAllTenants onTenants,
+            FactoryFor<IEmbeddingDefinitions> getDefinitions)
+        {
+            _onTenants = onTenants;
+            _getDefinitions = getDefinitions;
+        }
+
+        /// <inheritdoc/>
+        public async Task<IDictionary<TenantId, EmbeddingDefinitionComparisonResult>> DiffersFromPersisted(
+            EmbeddingDefinition definition,
+            CancellationToken token)
+        {
+            var results = new Dictionary<TenantId, EmbeddingDefinitionComparisonResult>();
+            await _onTenants.PerformAsync(async tenant =>
+            {
+                var definitions = _getDefinitions();
+                var tryGetDefinition = await definitions.TryGet(definition.Embedding, token).ConfigureAwait(false);
+                var comparisonResult = tryGetDefinition.Success switch
+                {
+                    true => DefinitionsAreEqual(definition, tryGetDefinition),
+                    false => EmbeddingDefinitionComparisonResult.Equal
```

Comment:
So if the definition isn't persisted, then `tryGetDefinition.Success == false` and we will return saying that it was `EmbeddingDefinitionComparisonResult.Equal`? This feels a bit wrong, but I'm not sure if there is a big enough reason to have a third case, like `EmbeddingDefinitionComparisonResult.NotInitialized` or something. Now it just feels a bit wonky that a method `DiffersFromPersisted` returns equal, when the passed definition differs from persisted (which doesn't exist).

Reply:
Yeah I know. It's how the projections stuff does it now. There is no grand reason behind this. I think that we could fix this by changing the interface name. I think that we should note this and fix this together with doing the same thing for projections